processing/src/main/java/io/druid/query/groupby/epinephelinae/ConcurrentGrouper.java

```diff
@@ -94,6 +96,8 @@ public ConcurrentGrouper(
           )
       );
     }
+
+    this.entryComparator = keySerdeFactory.factorize().entryComparator();
```

Comment:
How does this sound:

- Change `KeyComparator comparator();` to `KeyComparator bufferComparator();`
- Change `Comparator<Entry<T>> entryComparator();` to `Comparator<T> objectComparator();` and move it to the KeySerdeFactory (or put it on both KeySerdeFactory and KeySerde), so we don't need to factorize and throw away a keySerde to get an object comparator.
- Have Groupers.mergeIterators accept the `Comparator<KeyType>` and then create a `Comparator<Entry<KeyType>>` out of that. I think the wrapping here will have an unnoticeable cost and will make the code nicer, by making the comparators all operate on keys.